### PR TITLE
Add detailed instructions for running the Java hello world sample.

### PR DESCRIPTION
**Pull request by tswast**

### CODE REVIEW

File: java/hello-world/README.md [hunk 1]
```diff
@@ -3,20 +3,132 @@
 This is a simple application that demonstrates using the native HBase API
 to connect to and interact with Cloud Bigtable.
```
Comment (lesv) [inline java/hello-world/README.md]: Should there be a table of contents?

File: java/hello-world/README.md [hunk 2]
```diff
@@ -3,20 +3,132 @@
...
+
+## Before you begin
+
+This sample assumes you have [Java 8][java8] installed.
+
+[java8]: http://www.oracle.com/technetwork/java/javase/downloads/
+
+### Installing Maven
+
+These samples use the [Apache Maven][maven] build system. Before getting
```
Comment (lesv) [inline java/hello-world/README.md]: I think we usually talk about `brew` for mac, and `apt-get install` for many linux's.

File: java/hello-world/README.md [hunk 3]
```diff
@@ -3,20 +3,132 @@
...
+application.
+
+
+## Running the application
+
+Set the following environment variables or replace them with the appropriate
+values in the `mvn` commands. Set:
+
++   `GCLOUD_PROJECT` to the project ID,
++   `BIGTABLE_CLUSTER` to the Bigtable cluster ID,
```
Comment (lesv) [inline java/hello-world/README.md]: Changing soon

File: java/hello-world/README.md [hunk 4]
```diff
@@ -3,20 +3,132 @@
...
++   `BIGTABLE_ZONE` to the Bigtable compute zone (example: us-central1-b).
+
+Build and run the sample using Maven.
+
+    mvn package
+    mvn exec:java -Dbigtable.projectID=${GCLOUD_PROJECT} \
```
Comment (lesv) [inline java/hello-world/README.md]: Working on this now, will be -Dbigtable.projectID=... & -Dbigtable.instanceID=  removing clusterID & zone

**Comment by lesv** [comment]
a few changes, please update branch as well.

File: java/hello-world/README.md (hunk 2 above)
Comment (tswast) [inline java/hello-world/README.md]: I've found that the version of Maven is too far out-of-date when using `apt-get install`. We don't mention it here: https://cloud.google.com/appengine/docs/java/tools/maven#installing_maven_31

File: java/hello-world/README.md (hunk 1 above)
Comment (tswast) [inline java/hello-world/README.md]: Added. I used a tool called doctoc. I'm considering making a bot just for README updates like this. There are plenty of areas where we could use templates for repeated steps.

File: java/hello-world/README.md (hunk 4 above)
Comment (tswast) [inline java/hello-world/README.md]: Acknowledged.

File: java/hello-world/README.md (hunk 2 above)
Comment (lesv) [inline java/hello-world/README.md]: Yep - I'm trying to move all the samples to 3.3.9

**Comment by lesv** [comment]
LGTM

**Pull request merged**